app/src/main/java/org/mozilla/fenix/home/WallpapersObserver.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home
+
+import android.view.View
+import android.widget.ImageView
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import mozilla.components.lib.state.Store
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.components.appstate.AppState
+import org.mozilla.fenix.ext.scaleToBottomOfView
+import org.mozilla.fenix.theme.ThemeManager
+import org.mozilla.fenix.wallpapers.Wallpaper
+import org.mozilla.fenix.wallpapers.WallpapersUseCases
+
+/**
+ * [LifecycleObserver] that will immediately start observing the store for wallpapers updates
+ * and apply them to the passed in [wallpaperImageView].
+ *
+ * @param appStore Holds the details abut the current wallpaper.
+ * @param wallpapersUseCases Used for interacting with the wallpaper feature.
+ * @param themeManager Holds the details about the current theme.
+ * @param wallpaperImageView Serves as the target when applying wallpapers.
+ */
+class WallpapersObserver(
+    private val appStore: AppStore,
+    private val wallpapersUseCases: WallpapersUseCases,
+    private val themeManager: ThemeManager,
+    private val wallpaperImageView: ImageView,
+) : DefaultLifecycleObserver {
+    @VisibleForTesting
+    internal var observeWallpapersStoreSubscription: Store.Subscription<AppState, AppAction>? = null
+    @VisibleForTesting
+    internal var wallpapersScope = CoroutineScope(Dispatchers.IO)
+
+    init {
+        if (shouldEnableWallpaper()) {
```

Comment:
Maybe we should not even initialize `WallpapersObserver` in the HomeFragment if `shouldEnableWallpaper` is false

Reply:
Wanted to have all the logic about wallpapers here but agree that not interacting with this object if there's no need for it seems cleaner.